### PR TITLE
Support AWS KMS key

### PR DESCRIPTION
**Pull request by stafot**
#### Summary
Support AWS KMS key

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-6393

#### Release Note

```release-note
Support custom AWS KMS key for EBS encryption
```

### CODE REVIEW

**Comment by gabrieljackson** [comment]
I have a general question to check that my understanding is correct about how the KMS key will be used. I see that we are using it to perform EBS encryption in this PR, but are there going to be future uses for the same KMS key?

**Comment by stafot** [comment]
We might want to perform EBS encryption on custom node groups in the future but I don't think this will happen anytime soon. 

Let me explain the purpose of this PR. 
Right now we already encrypt EBS by default using a KMS key. 
With this PR we can change and accept any other custom KMS to encrypt data of volumes at rest. SImilar tooling we have in other components (S3, RDS) implemented in other infrastructure modules.

**Review by gabrieljackson — APPROVED** [review]
LGTM, just one question below.

File: internal/provisioner/kops_utils.go [hunk 1]
```diff
@@ -86,3 +87,22 @@ func updateKopsInstanceGroupValue(kops *kops.Cmd, kopsMetadata *model.KopsMetada
...
+
+func updateWorkersKopsInstanceGroupValue(kops *kops.Cmd, kopsMetadata *model.KopsMetadata, value string) error {
+
+	instanceGroups, err := kops.GetInstanceGroupsJSON(kopsMetadata.Name)
+	if err != nil {
+		return errors.Wrap(err, "failed to get instance groups")
+	}
+
+	for _, ig := range instanceGroups {
+		if strings.Contains(ig.Metadata.Name, "nodes-") {
```
[the remark this answers is not in the document]
Comment (stafot) [reply internal/provisioner/kops_utils.go:99]: @gabrieljackson Yes, this is on purpose.

**Comment by stafot** [comment]
@mattermost/cloud-sre PTAL before proceed to merge this

**Review by angeloskyratzakos — APPROVED** [review]

**Pull request merged**